server/svix-server/src/v1/utils.rs

```diff
@@ -51,13 +84,28 @@ pub trait ModelIn {
 pub trait ModelOut {
     fn id_copy(&self) -> String;
 
-    fn list_response<T: ModelOut + Clone>(mut data: Vec<T>, limit: usize) -> ListResponse<T> {
+    fn list_response<T: ModelOut + Clone>(
+        mut data: Vec<T>,
+        limit: usize,
+        is_prev_iter: bool,
+    ) -> ListResponse<T> {
         let done = data.len() <= limit;
-        data.truncate(limit);
+
+        if is_prev_iter {
+            if data.len() > limit {
+                data = data.drain(data.len() - limit..).collect();
+            }
+        } else {
+            data.truncate(limit);
+        }
```

Comment:
```suggestion
        if data.len() > limit {
             if is_prev_iter {
                data = data.drain(data.len() - limit..).collect();
             } else {
                data.truncate(limit);
            }
        }
```
Makes the behavior more clear.